### DISCUSSION TITLE
[css-tree] Incorrect types for AST nodes

### DISCUSSION THREAD

**Discussion by MMK21Hub** [DefinitelyTyped/DefinitelyTyped#58334] [Issues with a @types package] [answered]
I'm using this code snippet from the examples, and it works without Typescript. However, Typescript is throwing an error. It looks like something is incorrect in `DSNode`, or `CssNodeCommon` should be used instead. Here's the code I'm using:
```ts
if (node.type === "ClassSelector" && node.name === "example") {
  node.name = "hello"
}
```
```
This condition will always return 'false' since the types '"Group" | "AtKeyword" | "Comma" | "Function" | "Keyword" | "Multiplier" | "Property" | "String" | "Token" | "Type"' and '"ClassSelector"' have no overlap.
```

I ran a simple test in the JS console to verify that `node.type` can have other values:
![image](https://user-images.githubusercontent.com/50421330/150383124-e831fa0f-0980-41e5-9447-3ba7b1811dee.png)

[a comment by an automated account is not shown]

**Comment by pyoor** [comment, marked as answer]
@MMK21Hub Can you provide your entire code sample?  DSNode should only be used with the [CSS value definition-syntax parser](https://github.com/csstree/csstree/tree/master/lib/definition-syntax).

**Reply by MMK21Hub** [reply]
Thanks for the quick response! It looks like I used the wrong type when trying to fix the issue: I've updated my `import` and it works correctly now :facepalm:
Sorry for wasting your time!
